main-thread.lisp

```diff
@@ -72,22 +72,25 @@
   (setf *main-runner-running-p* T)
   (restart-case
       (handler-bind ((error #'main-error-handler))
-        (unwind-protect
-             (loop (when (bt:with-lock-held (*task-lock*)
-                           (bt:condition-wait *task-cvar* *task-lock* :timeout 0.5))
-                     (let ((tasks (bt:with-lock-held (*task-lock*)
-                                    (shiftf *task-queue* (make-array 32 :fill-pointer 0)))))
+        (let (tasks)
+          (unwind-protect
+               (loop (when (bt:with-lock-held (*task-lock*)
+                             (bt:condition-wait *task-cvar* *task-lock* :timeout 0.5))
+                       (setf tasks (bt:with-lock-held (*task-lock*)
+                                     (shiftf *task-queue* (make-array 32 :fill-pointer 0))))
```

Comment:
Minor nitpick: would probably be better to use a single *back-task-queue* that is rotated here to avoid allocating a new one, but I doubt it matters much.

Reply:
Yeah, that would be better. But it was already as such, so I didn't bother much for today. I might get back to it later.